File: Leetcode/leetcode.py
def numberOfSubarrays(self, nums, k):
    i=0
    j=0
    count=0
    ans=0
    while(i<len(nums)):
        if nums[i]%2==1:
            count+=1
        if count==k:
            temp=i
            while(i<len(nums) and nums[i]%2==0):
                i+=1
            left=i-temp
            while(i<len(nums) and nums[i]%2==1):
                i+=1
            right=i-temp
            ans+=left*right
            count-=1
        else:
            i+=1

File: Leetcode/leetcode.js
var numberOfSubarrays = function(nums, k) {
    let niceCount = 0;
    let foundOdd = 0;
    for(let i=0; i< nums.length; i++){
        for(let j=i; j< nums.length; j++){
            if(nums[j] % 2 == 1) {
                foundOdd++;
            }
            if(foundOdd > k) {
                break;
            }

            if(foundOdd == k){
                console.log(i,j)
                niceCount++
            }
        }
        foundOdd = 0
    }
    return niceCount;
};